src/lambda/commands/deploySamApplication.ts

```diff
@@ -13,137 +13,200 @@ import * as nls from 'vscode-nls'
+    destinationStackName: string
+    parameterOverrides: Map<string, string>
+}
+
 export async function deploySamApplication(
     {
-        invoker = new DefaultValidatingSamCliProcessInvoker({}),
+        samCliContext = getSamCliContext(),
+        channelLogger,
         ...restParams
     }: {
-        invoker?: SamCliProcessInvoker
-        outputChannel: vscode.OutputChannel
+        samCliContext?: SamCliContext
+        channelLogger: ChannelLogger,
         regionProvider: RegionProvider
     },
     awsContext: Pick<AwsContext, 'getCredentialProfileName'>
-) {
-    const args: SamDeployWizardResponse | undefined = await new SamDeployWizard(restParams.regionProvider).run()
-    if (!args) {
-        return
+): Promise<void> {
+    try {
+        const profile: string | undefined = awsContext.getCredentialProfileName()
+        if (!profile) {
+            throw new NoActiveCredentialError()
+        }
+
+        throwAndNotifyIfInvalid(await samCliContext.validator.detectValidSamCli())
+
+        const deployWizardResponse: SamDeployWizardResponse | undefined = await new SamDeployWizard(
+            restParams.regionProvider
+        ).run()
+
+        if (!deployWizardResponse) {
+            return
+        }
+
+        const deployParameters: DeploySamApplicationParameters = {
+            deployRootFolder: await makeTemporaryToolkitFolder('samDeploy'),
+            destinationStackName: deployWizardResponse.stackName,
+            packageBucketName: deployWizardResponse.s3Bucket,
+            parameterOverrides: deployWizardResponse.parameterOverrides,
+            profile,
+            region: deployWizardResponse.region,
+            sourceTemplatePath: deployWizardResponse.template.fsPath,
+        }
+
+        const deployApplicationPromise = buildPackageDeploy({
+            deployParameters,
+            channelLogger,
+            invoker: samCliContext.invoker,
+        }).then(async () =>
+            await del(deployParameters.deployRootFolder, {
+                force: true
+            })
+        )
+
+        vscode.window.setStatusBarMessage(
+            localize(
+                'AWS.samcli.deploy.statusbar.message',
+                '$(cloud-upload) Deploying SAM Application to {0}...',
+                deployWizardResponse.stackName,
+            ),
+            deployApplicationPromise
+        )
+    } catch (err) {
+        const error = err as Error
+        channelLogger.logger.error(error)
+
+        channelLogger.channel.show(true)
+        channelLogger.error(
+            'AWS.samcli.deploy.general.error',
+            'An error occurred while deploying a SAM Application. Check the logs for more information.'
+        )
     }
+}
+
+function getBuildRootFolder(deployRootFolder: string): string {
+    return path.join(deployRootFolder, 'build')
+}
+
+function getBuildTemplatePath(deployRootFolder: string): string {
+    return path.join(getBuildRootFolder(deployRootFolder), 'template.yaml')
```

Comment:
Will this always be the generated template? If not, we need to support `template.yml` as well.

Reply:
I'll make a note in the comments to keep an eye on this